lib/rspec/mocks/order_group.rb

```diff
@@ -41,7 +41,8 @@ def handle_order_constraint(expectation)
       end
 
       def verify_invocation_order(expectation)
-        expectation.raise_out_of_order_error unless expectations_invoked_in_order?
+        message = @invocation_order[@index] ? @invocation_order[@index].message : nil
```

Comment:
I think this would read better as `message = @invocation_order[@index].message if @invocation_order[@index]`